_protected/app/system/modules/mail/models/MailModel.php

```diff
@@ -73,7 +73,29 @@ public function readSentMsg($iSender, $iMessageId)
 
         return $rStmt->fetch(\PDO::FETCH_OBJ);
     }
+    
+    public function getMsgHistory($iRecipient, $iSender)
+    {
+        $rStmt = Db::getInstance()->prepare(
+            'SELECT msg.*, m.profileId, m.username, m.firstName FROM' . Db::prefix(DbTableName::MESSAGE) .
+            'AS msg LEFT JOIN ' . Db::prefix(DbTableName::MEMBER) . 'AS m ON msg.sender = m.profileId
+            AND msg.recipient OR msg.sender = m.profileId
+            WHERE msg.recipient = :recipient
+            AND msg.sender = :sender OR msg.sender = :recipient
+            AND NOT FIND_IN_SET(\'recipient\', msg.toDelete)
+            AND NOT FIND_IN_SET(\'sender\', msg.toDelete) LIMIT 25'
```

Comment:
Probably need to add `ORDER BY sendDate DESC` here, before the LIMIT clause.

Reply:
Yes you are right, good idea.